### PR TITLE
DTSPO-13430: Switch chart library to helmunittest

### PR DESCRIPTION
**Pull request by DanielKChristou**
+ Added snapshot tests for the pre-existing templates
+ Added unit tests for the hpa template

**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-13430

### Change description ###
Created snapshot tests for templates within chart-library. 
Created a unit test for the HPA template to serve as a example for future tests. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: library/Chart.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ description: A Helm chart for Kubernetes
 # Library charts provide useful utilities or functions for the chart developer. They're included as
 # a dependency of application charts to inject those utilities and functions into the rendering
 # pipeline. Library charts do not define any templates and therefore cannot be deployed.
-type: library
+type: application
```
Comment (timja) [inline library/Chart.yaml:12]: this shouldn't be checked in, it needs to be library

File: library/tests/unit-tests/hpa_test.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          path: spec.metrics
+          content:
+            resource:
+              name: cpu
+              target:
+                averageUtilization: 80
+                type: Utilization
+            type: Resource
+
+  - it: should create parameters for memory autoscaling when memory autoscaling is disabled.
```
Comment (timja) [inline library/tests/unit-tests/hpa_test.yaml:38]:
```suggestion
  - it: should create parameters for memory autoscaling when memory autoscaling is enabled.
```

File: library/tests/unit-tests/hpa_test.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          path: spec.metrics
+          content:
+             resource:
+                name: memory
+                target:
+                  averageUtilization: 80
+                  type: Utilization
+             type: Resource
+
+  - it: should not create parameters for memory scaling
```
Comment (timja) [inline library/tests/unit-tests/hpa_test.yaml:55]:
```suggestion
  - it: should not create parameters for memory scaling when memory autoscaling is disabled
```

File: library/tests/unit-tests/hpa_test.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      - isKind:
+          of: HorizontalPodAutoscaler
+      - notContains:
+          path: spec.metrics
+          content:
+            resource:
+              name: memory
+              target:
+                averageUtilization: 80
+                type: Utilization
```
Comment (timja) [inline library/tests/unit-tests/hpa_test.yaml:69]: I don't think we need to assert the value when we are checking it's not present
```suggestion
```

File: azure-pipelines.yaml [hunk 5]
```diff
@@ -28,8 +28,11 @@ jobs:
       - task: HelmInstaller@1
         displayName: 'Install Helm'
         inputs:
-          helmVersionToInstall: 3.0.2
+          helmVersionToInstall: 3.12.0
+      # Install helm unittest plugin
```
Comment (timja) [inline azure-pipelines.yaml:32]:
```suggestion
```

File: azure-pipelines.yaml [hunk 6]
```diff
@@ -28,8 +28,11 @@ jobs:
...
+      - script: helm plugin install https://github.com/helm-unittest/helm-unittest.git
       - script: sudo snap install yq
+      # Run tests
```
Comment (timja) [inline azure-pipelines.yaml:35]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -344,10 +344,13 @@ Adds a Kubernetes service based on the pod's properties.
 As library charts are not installable, you can use [this script](tests/test-templates.sh) to generate templates and validate your changes.
 
 ### Changing existing templates
- - Add the change you want to see in [results](tests/results/) 
- - Run the [script](tests/test-templates.sh) to see if the generated templates match.
- - Generated manifests can be installed on a cluster if you want to see they are working as expected.
- - Make sure language specific cases are covered in the tests [see](ci-values-lang.yaml) 
+- Add the change you want to see in the code and run [this script](tests/test-templates.sh)
```
Comment (timja) [inline README.md:347]:
```suggestion
- Add the change you want to see in the code and run [the tests](tests/test-templates.sh)
```

**Review by timja — APPROVED** [review]
LGTM, minor suggestions, the comments weren't useful to me. 

e.g. 
install unit test plugin
helm installl unittest plugin

**Review by reespozzi — APPROVED** [review]

**Pull request merged**